google/api_core/retry_async.py

```diff
@@ -74,23 +94,23 @@ async def check_if_exists():
 async def retry_target(
     target, predicate, sleep_generator, timeout=None, on_error=None, **kwargs
```

Comment:
the streaming versions have type declarations. Do you want to have type declarations in the non-stream versions?